analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.cs

```diff
@@ -193,4 +193,41 @@ public void M3()
             double doTheThing(int divisor, int dividend) => divide(dividend, divisor);  // Noncompliant
         }
     }
+
+    // See https://github.com/SonarSource/sonar-dotnet/issues/3879
+    class NullableParam
+    {
+        void NotNullableParamVoid(int a, int b) // Secondary [D,E,F]
+        {
+            // Do nothing
+        }
+
+        void NullableParamValueVoid(int a, int? b)
+        {
+            if (b.HasValue)
+            {
+                NotNullableParamVoid(b.Value, a); // Noncompliant [D] FN
+                NotNullableParamVoid(a, b.Value); // Compliant [D]
+            }
+        }
+
+        void NullableParamCastVoid(int a, int? b)
```

Comment:
Let's use `Nullable<int>` as well.

Reply:
ok, but why do you preffer `Nullable<int>?`

Reply:
I don't prefer `Nullable<int>`, I wanted to have a test case for `int?` as well as `Nullable<int>` to document the behavior and to be sure that it works for both. Can you revert one of them to `int?` please?